service/devdoc/registry_requirements.md

```diff
@@ -370,17 +370,234 @@ Request-Id: <guid>
+**SRS_NODE_IOTHUB_REGISTRY_18_009: [** The `addConfiguration` method shall set `configuration.schemaVersion` to the constant `currentConfigurationSchemaVersion` if it is not already set. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_010: [** The `addConfiguration` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+PUT /configurations/<encodeURIComponent(configuration.id)>?api-version=<version> HTTP/1.1
+Authorization: <sharedAccessSignature>
+Content-Type: application/json; charset=utf-8
+Request-Id: <guid>
+
+<configuration>
+```
+**]**
+
+### getConfiguration(configurationId: string, done: Registry.ResponseCallback): void;
+
+**SRS_NODE_IOTHUB_REGISTRY_18_011: [** The `getConfiguration` method shall throw a `ReferenceError` exception if `configurationId` is falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_012: [** The `getConfiguration` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+GET /configurations/<encodeURIComponent(configurationId)>?api-version=<version> HTTP/1.1
+Authorization: <sharedAccessSignature>
+Request-Id: <guid>
+```
+**]**
+
+### getConfigurations(done: Registry.ResponseCallback): void;
+
+**SRS_NODE_IOTHUB_REGISTRY_18_013: [** The `getConfigurations` method shall throw a `ReferenceError` exception if `done` is falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_014: [** The `getConfigurations` method shall construct an HTTP request using information supplied by the caller, as follows:
+```
+GET /configurations?api-version=<version> HTTP/1.1
+Authorization: <sharedAccessSignature>
+Request-Id: <guid>
+```
+**]**
+
+### updateConfiguration(configuration: Configuration, done: Registry.ResponseCallback): void;
+### updateConfiguration(configuration: Configuration, forceUpdate: boolean, done: Registry.ResponseCallback): void;
+### updateConfiguration(configuration: Configuration, forceUpdateOrDone: boolean | Registry.ResponseCallback, done?: Registry.ResponseCallback): void;
+
+**SRS_NODE_IOTHUB_REGISTRY_18_015: [** The `updateConfiguration` method shall throw a `ReferenceError` exception if `configuration` or `done` is falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_016: [** The `updateConfiguration` method shall throw an `ArgumentError` exception if `forceUpdate` is falsy and `configuration.etag` is also falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_017: [** The `updateConfiguration` method shall throw an `ArgumentError` exception if `configuration.id` is falsy. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_018: [** The `updateConfiguration` method shall set ``configuration.schemaVersion` to the constant `currentConfigurationSchemaVersion` if it is not already set. **]**
+
+**SRS_NODE_IOTHUB_REGISTRY_18_020: [** If `forceUpdate` is not truthy, the `updateConfigurationMethod` shall put the `etag` parameter into the `If-Match` header value. **]**
```

Comment:
>If-Match` header value. **] [](start = 144, length = 27)

And if there is no etag property?

Reply:
SRS_NODE_IOTHUB_REGISTRY_18_016 ?

---
In reply to: [176863077](https://github.com/Azure/azure-iot-sdk-node/pull/256#discussion_r176863077) [](ancestors = 176863077)

Reply:
Pierre is right.  That should fail above because of 18_016


---
In reply to: [176890424](https://github.com/Azure/azure-iot-sdk-node/pull/256#discussion_r176890424) [](ancestors = 176890424,176863077)